RELEASE_PROCESS.md

```diff
@@ -101,19 +88,6 @@ The git SHA of the build you are testing can be found in
 
 ### Preparing for the release
 
-- [ ] Check that omnibus_overrides.rb has the correct version given
-  the type of changes in this release.
-
-- [ ] Download the previous release of Chef Server (.deb).
-
-- [ ] Run `./dev/scripts/update-changelog.sh /path/to/deb/from/previous/step`
-
-- [ ] Grab the section on the latest version from `NEW_CHANGELOG.md` and paste
-  it at the top of `CHANGELOG.md`.
-
-- [ ] Copy the components section from `MODIFIED_COMPONENTS_CHANGELOG.md` and
```

Comment:
I think we could live without the "Contributors" section; however, I see a couple of advantages: (1) I think it is nice to call out contributors in a more prominent section and (2) it handles cases where multiple authors worked on the same PR nicely.